docs/source/api/jobs.rst

```diff
@@ -21,78 +21,470 @@
+
+:Auth. Required: Yes
+:Roles Required: "operations" or "admin"\ [#tenancy]_
+:Response Type:  Object
+
+Request Structure
+-----------------
+.. table:: Query Parameters
+
+	+------+----------+--------------------------------------------------------------------------------+
+	| Name | Required | Description                                                                    |
+	+======+==========+================================================================================+
+	| id   | yes      | The integral, unique identifier of the content invalidation job being modified |
+	+------+----------+--------------------------------------------------------------------------------+
+
+:assetUrl: A regular expression - matching URLs will be operated upon according to ``keyword``
+
+	.. note:: Unlike in the payloads of POST_ and PATCH_ requests to this endpoint, this must be a **full** URL regular expression, as it is **not** combined with the :ref:`ds-origin-url` of the :term:`Delivery Service` identified by ``deliveryService``.
+
+:createdBy:       The username of the user who initiated the job\ [#readonly]_
+:deliveryService: The :ref:`ds-xmlid` of the :term:`Delivery Service` on which this job operates\ [#readonly]_ - unlike POST_ and PATCH_ request payloads, this cannot be an integral, unique identifier
+:id:              An integral, unique identifier for this job\ [#readonly]_
+:keyword:         A keyword that represents the operation being performed by the job. It can have any (string) value, but the only value with any meaning to Traffic Control is:
+
+	PURGE
+		This job will prevent caching of URLs matching the ``assetUrl`` until it is removed (or its Time to Live expires)
+
+:parameters: A string containing space-separated key/value pairs - delimited by colons (:kbd:`:`\ s) representing parameters associated with the job. In practice, any string can be passed as a job's ``parameters``, but the only value with meaning is a single key/value pair indicated a :abbr:`TTL (Time To Live)` in hours in the format :file:`TTL:{hours}h`, and any other type of value may cause components of Traffic Control to work improperly or not at all.
+:startTime:  This can be a string in the legacy ``YYYY-MM-DD HH:MM:SS`` format, or a string in :rfc:`3339` format, or a string representing a date in the same non-standard format as the ``last_updated`` fields common in other API responses, or finally it can be a number indicating the number of milliseconds since the Unix Epoch (January 1, 1970 UTC). This **must** be in the future, but only by no more than two days.
+
+.. code-block:: http
+	:caption: Request Example
+
+	PUT /api/1.4/jobs?id=3 HTTP/1.1
+	Host: trafficops.infra.ciab.test
+	User-Agent: python-requests/2.20.1
+	Accept-Encoding: gzip, deflate
+	Accept: */*
+	Connection: keep-alive
+	Cookie: mojolicious=...
+	Content-Length: 188
+	Content-Type: application/json
+
+	{
+		"assetUrl": "http://origin.infra.ciab.test/.*",
```

Comment:
this is related to one of my earlier comments. if we allow assetUrl to be origin fqdn + path, can't somebody create an invalidation content request for an origin that is not theirs?

Reply:
If you _don't_ allow that, then Delivery Services with multiple origins can potentially be incapable of content revalidations. It's not ideal, but changing it means redefining what a job does, which potentially involves a loss of functionality.

This behavior is consistent with the current method of job creation under `/user/current/jobs`, so any 'fix' here wouldn't be effective.

Reply:
ok, so is there some type of check that when i'm doing a PUT to a job, that the assetUrl lines up with one of the origins of the delivery service? if not, this would expose a security hole imo.

Reply:
So, Job creation specifically combines the Regex with the Delivery Service's primary origin from the origin table - that's the only one I'll be checking, then.